urllib3/poolmanager.py

```diff
@@ -42,12 +42,39 @@
     'HTTPSPoolKey', HTTPPoolKey._fields + SSL_KEYWORDS
 )
 
-# The default pool key for each scheme; each PoolManager makes a copy of this
-# dictionary so they can be configured globally here, or individually on the
-# instance.
-pool_keys_by_scheme = {
-    'http': HTTPPoolKey,
-    'https': HTTPSPoolKey,
+
+def default_key_normalizer(request_context, key_class):
```

Comment:
Hmmm I imagine the name/behaviour of this function changing, let's make it `_`-prefixed just in case.